File: typescript/base-hmi-types/src/screens/controls/HmiAuditTrailControl.ts
import { HmiProperty } from "../base.js";
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiAuditTrailFieldPresentation } from "./HmiAuditTrailFieldPresentation.js";
import { HmiAuditTrailViewKind } from "./HmiAuditTrailViewKind.js";

export class HmiAuditTrailControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiAuditTrailControl;
  }

  viewKind = HmiAuditTrailViewKind.List;
  showHeader?: HmiProperty<boolean>;
  linesPerEntry?: HmiProperty<number>;
  wordWrap?: HmiProperty<boolean>;
  wrapAround?: HmiProperty<boolean>;
  receiveSelectionFrom?: string;
  readonly fields: HmiAuditTrailFieldPresentation[] = [];
}

File: src/BaseHmiTypes/Screens/Controls/HmiAuditTrailControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAuditTrailControl : HmiControlWindowBase
{
    public HmiAuditTrailControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiAuditTrailControl;
    }

    public HmiAuditTrailViewKind ViewKind { get; set; }

    public HmiProperty<bool>? ShowHeader { get; set; }

    public HmiProperty<int>? LinesPerEntry { get; set; }

    public HmiProperty<bool>? WordWrap { get; set; }

    public HmiProperty<bool>? WrapAround { get; set; }

    public string? ReceiveSelectionFrom { get; set; }

    public HmiProperty<HmiColor>? SelectionBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? SelectionForegroundColor { get; set; }

    public IList<HmiAuditTrailFieldPresentation> Fields { get; } = new List<HmiAuditTrailFieldPresentation>();
}
